core/src/main/java/org/jivesoftware/sparkimpl/profile/AvatarPanel.java

```diff
@@ -214,8 +209,16 @@ public void finished() {
                     return;
                 }
                 */
-                setAvatar(new ImageIcon(avatarImage));
-                avatarFile = selectedFile;
+                //convert BufferedImage to bytes
+                try {
+                    ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
But it is not required. Just found why here: http://stackoverflow.com/questions/2330569/closing-a-bytearrayoutputstream-has-no-effect